src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -163,24 +164,65 @@ private static ScriptHostConfiguration CreateScriptHostConfiguration(WebHostSett
 
             // If running on Azure Web App, derive the host ID from the default subdomain
             // Otherwise, derive it from machine name and folder name
-            string hostId = _settingsManager.AzureWebsiteDefaultSubdomain
-                ?? $"{Environment.MachineName}-{Path.GetFileName(Environment.CurrentDirectory)}";
+            string hostId = MakeValidHostID(
```

Comment:
We want to make sure that, on Azure, we use what is provided by the environment with no modification (and it should be safe, so the extra processing isn't needed), so we can probably change this to:
`_settingsManager.AzureWebsiteDefaultSubdomain ?? MakeValidHostID($"{Environment.MachineName}-{Path.GetFileName(Environment.CurrentDirectory)}");`